tasks/main.yml

```diff
@@ -0,0 +1,69 @@
+---
+- name: add the rabbitmq repo key
+  apt_key: data="{{ lookup('file', 'rabbitmq.asc') }}" state=present
+
+- name: add the erlang repo key
+  apt_key: data="{{ lookup('file', 'erlang_solutions.asc') }}" state=present
+  when: ansible_distribution_release == 'precise'
+
+- name: add rabbitmq repo
+  apt_repository:
+    repo: "deb http://www.rabbitmq.com/debian/ testing main"
+    state: present
+    update_cache: yes
+  # Note: the update_cache above should only run if we changed the system's repository config
+
+# Rabbitmq-server now needs a newer Erlang than the repos provide in
+# 12.04.  We can get that from erlang-solutions.com for 12.04, but not
```

Comment:
Since 12.04 is now out of support and this role hasn't yet been adoptable for new projects, I think the two "when.*precise" tasks can be removed.